zerver/lib/export.py

```diff
@@ -305,16 +305,41 @@ def write_data_to_file(output_file: Path, data: Any) -> None:
     with open(output_file, "w") as f:
         f.write(ujson.dumps(data, indent=4))
 
+def to_dict(instance: Any, fields: Optional[List[Field]]= None,
+            exclude: Optional[List[Field]]= None) -> Dict[str, Any]:
+    """
+    Modified form of model_to_dict function of django which converts non editable fields also.
+
+    Return a dict containing the data in ``instance`` suitable for passing as
+    a Form's ``initial`` keyword argument.
+
+    ``fields`` is an optional list of field names. If provided, return only the
+    named.
+
+    ``exclude`` is an optional list of field names. If provided, exclude the
+    named from the returned dict, even if they are listed in the ``fields``
+    argument.
+    """
+    opts = instance._meta
+    data = {}
+    for f in chain(opts.concrete_fields, opts.private_fields, opts.many_to_many):
+        if fields and f.name not in fields:
+            continue
+        if exclude and f.name in exclude:
+            continue
+        data[f.name] = f.value_from_object(instance)
+    return data
```

Comment:
Can you explain precisely why you needed this modified `to_dict` field?  I see its intended to handle "non editable fields"; what non-editable fields are present in FillState?  A traceback for the error you're working around might be helpful.

Reply:
(Also, a comment just before whichever line(s) we changed from the original implementation would be helpful, e.g. "# Zulip change: ...").

Reply:
Aside from this question, the `FillState` export commit looks reasonable to me.

Reply:
@timabbott According to Django any field with auto_now=true will inherit editable=false too so in FillState model we have last_modified that has auto_now. When model_to_dict was used it had a check on non-editable fields. In to_dict function that check is removed. Although I tried to investigate why model_to_dict had that check to avoid non-editable field but got no result.

Reply:
OK, interesting.  It's possible the right answer is that we should just remove the `last_modified` field; I'm not sure what it's used for.

Reply:
I think the right answer here is to just remove that column as unused clutter rather than copying this code to work around it.

Reply:
I just merged 9ac3e1099c60e7267a0fe307919b46916648c25d removing it.  Thanks for bringing this to my attention!